.ci/releasePipeline.groovy

```diff
@@ -279,19 +279,34 @@ void signWithSignService(Map args = [source: null, version: null, destDir: '.',
 void createContainerAndPushToStorage(Map args = [version: null, tagLatest: false, registrySecretsPath: null, registry: null]) {
     stage("Publish container: registry=${args.registry}, version=${args.version}") {
         withEnv(["version=${args.version}"]) {
-            withVault(vaultSecrets: [[path        : "${args.registrySecretsPath}",
-                                      secretValues: [[envVar: 'registry', vaultKey: 'registry', isRequired: true],
-                                                     [envVar: 'repo', vaultKey: 'repo', isRequired: true],
-                                                     [envVar: 'username', vaultKey: 'username', isRequired: true],
-                                                     [envVar: 'password', vaultKey: 'password', isRequired: true]]]]) {
+            withVault(vaultSecrets: [
+                [
+                    path        : "${args.registrySecretsPath}",
+                    secretValues: [
+                        [envVar: 'registry', vaultKey: 'registry', isRequired: true],
+                        [envVar: 'repo', vaultKey: 'repo', isRequired: true],
+                        [envVar: 'username', vaultKey: 'username', isRequired: true],
+                        [envVar: 'password', vaultKey: 'password', isRequired: true]
+                    ]
+                ],
+                [
+                    path        : "keptn-jenkins/monaco/cosign",
+                    secretValues: [
+                        [envVar: 'cosign_key', vaultKey: 'cosign.key', isRequired: true],
+                        [envVar: 'cosign_password', vaultKey: 'cosign_password', isRequired: true]
+                    ]
+                ]
+            ]) {
                 script {
                     try {
                         sh 'docker login --username $username --password $password $registry'
                         sh 'DOCKER_BUILDKIT=1 make docker-container OUTPUT=./build/docker/monaco CONTAINER_NAME=$registry/$repo/dynatrace-configuration-as-code VERSION=$version'
-                        sh 'docker push $registry/$repo/dynatrace-configuration-as-code:$version'
+
+                        pushAndSign(imageName: '$registry/$repo/dynatrace-configuration-as-code:$version')
+
                         if (args.tagLatest) {
                             sh 'docker tag $registry/$repo/dynatrace-configuration-as-code:$version $registry/$repo/dynatrace-configuration-as-code:latest'
-                            sh 'docker push $registry/$repo/dynatrace-configuration-as-code:latest'
+                            pushAndSign(imageName: '$registry/$repo/dynatrace-configuration-as-code:latest')
```

Comment:
Technically it should not be necessary to sign here? 
Unless I misunderstand what cosign signs. 

This is just a `tag` of the same image - if I understand it correctly, cosign will sign the image based on the sha disgest. The 'latest' tag should point to exactly the same sha, so it should already be signed.

Reply:
![image](https://github.com/Dynatrace/dynatrace-configuration-as-code/assets/3740651/f8879ab2-7ff1-4356-bf51-0cf07e483d2d)

You can see on DockerHub that the images, although they have the same 'content,' have different digests, so we need to sign them each time.

Interestingly, not every registry behaves the same. For docker, it's a different digest, while in our internal registry, it's the same.

Reply:
Huh, interesting...
Does 2.1.1 already point to the same image for the latest? I can't recall when exactly I fixed that from being a dedicated image to just a tag

Reply:
Ah.. nope `2.1.1` and `latest` on Dockerhub are still two different images: 

```sh
> docker run dynatrace/dynatrace-configuration-as-code:latest version
Unable to find image 'dynatrace/dynatrace-configuration-as-code:latest' locally
latest: Pulling from dynatrace/dynatrace-configuration-as-code
Digest: sha256:b01bcb0cdf6efcd938b699822f2a9ffddafb79c5a1ea9d99dc51ddd70cb7838c
[...]
monaco version latest
```

```sh
> docker run dynatrace/dynatrace-configuration-as-code:2.1.1 version
Unable to find image 'dynatrace/dynatrace-configuration-as-code:2.1.1' locally
2.1.1: Pulling from dynatrace/dynatrace-configuration-as-code
Digest: sha256:00e60cb1fe78fe0dbf89db03ddbde5ddbe181fa8c86e48d878b4cad21b2e8e7a
[...]
monaco version 2.1.1
```

Reply:
Ah, you are right. I reverted the line to push only. Since we sign it before it will be signed and pushed :)